README
NAME
    Dist::Zilla::Plugin::DualBuilders - Allows use of Module::Build and
    ExtUtils::MakeMaker in a dzil dist

VERSION
      This document describes v1.001 of Dist::Zilla::Plugin::DualBuilders - released December 15, 2010 as part of Dist-Zilla-Plugin-DualBuilders.

DESCRIPTION
    This plugin allows you to specify ModuleBuild and MakeMaker in your
    Dist::Zilla dist.ini and select only one as your prereq. Normally, if
    this plugin is not loaded you will end up with both in your prereq list
    and this is obviously not what you want! Also, this will block both
    builders from running the testsuite twice.

            # In your dist.ini:
            [ModuleBuild]
            [MakeMaker] ; or [MakeMaker::Awesome], will work too :)
            [DualBuilders] ; needs to be specified *AFTER* the builders

ATTRIBUTES
  prefer
    Sets your preferred builder. This builder will be the one added to the
    prereqs. Valid options are: "make" or "build".

    The default is: build

  block_test
    This is a boolean value determining if we will block both testers from
    running the testsuite. If you have both builders loaded, you will run
    the testsuite twice! If you want this behavior, please set this value to
    false.

    The default is: true

SEE ALSO
    Please see those modules/websites for more information related to this
    module.

    *   Dist::Zilla

SUPPORT
    You can find documentation for this module with the perldoc command.

      perldoc Dist::Zilla::Plugin::DualBuilders

  Websites
    The following websites have more information about this module, and may
    be of help to you. As always, in addition to those websites please use
    your favorite search engine to discover more resources.

    *   Search CPAN

        <http://search.cpan.org/dist/Dist-Zilla-Plugin-DualBuilders>

    *   RT: CPAN's Bug Tracker

        <http://rt.cpan.org/NoAuth/Bugs.html?Dist=Dist-Zilla-Plugin-DualBuil
        ders>

    *   AnnoCPAN: Annotated CPAN documentation

        <http://annocpan.org/dist/Dist-Zilla-Plugin-DualBuilders>

    *   CPAN Ratings

        <http://cpanratings.perl.org/d/Dist-Zilla-Plugin-DualBuilders>

    *   CPAN Forum

        <http://cpanforum.com/dist/Dist-Zilla-Plugin-DualBuilders>

    *   CPANTS Kwalitee

        <http://cpants.perl.org/dist/overview/Dist-Zilla-Plugin-DualBuilders
        >

    *   CPAN Testers Results

        <http://cpantesters.org/distro/D/Dist-Zilla-Plugin-DualBuilders.html
        >

    *   CPAN Testers Matrix

        <http://matrix.cpantesters.org/?dist=Dist-Zilla-Plugin-DualBuilders>

  Internet Relay Chat
    You can get live help by using IRC ( Internet Relay Chat ). If you don't
    know what IRC is, please read this excellent guide:
    <http://en.wikipedia.org/wiki/Internet_Relay_Chat>. Please be courteous
    and patient when talking to us, as we might be busy or sleeping! You can
    join those networks/channels and get help:

    *   irc.perl.org

        You can connect to the server at 'irc.perl.org' and join this
        channel: #perl-help then talk to this person for help: Apocalypse.

    *   irc.freenode.net

        You can connect to the server at 'irc.freenode.net' and join this
        channel: #perl then talk to this person for help: Apocal.

    *   irc.efnet.org

        You can connect to the server at 'irc.efnet.org' and join this
        channel: #perl then talk to this person for help: Ap0cal.

  Bugs / Feature Requests
    Please report any bugs or feature requests by email to
    "bug-dist-zilla-plugin-dualbuilders at rt.cpan.org", or through the web
    interface at
    <http://rt.cpan.org/NoAuth/ReportBug.html?Queue=Dist-Zilla-Plugin-DualBu
    ilders>. I will be notified, and then you'll automatically be notified
    of progress on your bug as I make changes.

  Source Code
    The code is open to the world, and available for you to hack on. Please
    feel free to browse it and play with it, or whatever. If you want to
    contribute patches, please send me a diff or prod me to pull from your
    repository :)

    <http://github.com/apocalypse/perl-dist-zilla-plugin-dualbuilders>

      git clone git://github.com/apocalypse/perl-dist-zilla-plugin-dualbuilders.git

AUTHOR
    Apocalypse <APOCAL@cpan.org>

COPYRIGHT AND LICENSE
    This software is copyright (c) 2010 by Apocalypse.

    This is free software; you can redistribute it and/or modify it under
    the same terms as the Perl 5 programming language system itself.

    The full text of the license can be found in the LICENSE file included
    with this distribution.



Changes
Revision history for Dist::Zilla::Plugin::DualBuilders

1.001
	Released: 2010-12-15 18:54:27 UTC

	Add the ability to block both builders from running the testsuite twice

1.000
	Released: 2010-12-14 03:27:05 UTC

	Converted to Dist::Zilla for the release process!
	Tweaked some POD/error messages for clarity

0.03

	Update the prereq logic to work with Dist::Zilla's migration to CPAN META v2

0.02

	Ah, I forgot to munge the Makefile.PL/Build.PL to contain the proper builder...

0.01

	initial release


Build.PL

use strict;
use warnings;

use Module::Build 0.3601;


my %module_build_args = (
  'build_requires' => {
    'File::Find' => '0',
    'File::Temp' => '0',
    'Module::Build' => '0.3601',
    'Test::More' => '0.88'
  },
  'configure_requires' => {
    'Module::Build' => '0.3601'
  },
  'dist_abstract' => 'Allows use of Module::Build and ExtUtils::MakeMaker in a dzil dist',
  'dist_author' => [
    'Apocalypse <APOCAL@cpan.org>'
  ],
  'dist_name' => 'Dist-Zilla-Plugin-DualBuilders',
  'dist_version' => '1.001',
  'license' => 'perl',
  'module_name' => 'Dist::Zilla::Plugin::DualBuilders',
  'recommends' => {},
  'recursive_test_files' => 1,
  'requires' => {
    'Dist::Zilla::Role::AfterBuild' => '3.101461',
    'Dist::Zilla::Role::InstallTool' => '3.101461',
    'Dist::Zilla::Role::PrereqSource' => '3.101461',
    'Moose' => '1.03',
    'Moose::Util::TypeConstraints' => '1.01',
    'perl' => '5.006'
  },
  'script_files' => []
);


my $build = Module::Build->new(%module_build_args);

$build->create_build_script;


Makefile.PL

use strict;
use warnings;

BEGIN { require 5.006; }

use ExtUtils::MakeMaker 6.31;



my %WriteMakefileArgs = (
  'ABSTRACT' => 'Allows use of Module::Build and ExtUtils::MakeMaker in a dzil dist',
  'AUTHOR' => 'Apocalypse <APOCAL@cpan.org>',
  'BUILD_REQUIRES' => {
    'File::Find' => '0',
    'File::Temp' => '0',
    'ExtUtils::MakeMaker' => '6.31',
    'Test::More' => '0.88'
  },
  'CONFIGURE_REQUIRES' => {
    'ExtUtils::MakeMaker' => '6.31'
  },
  'DISTNAME' => 'Dist-Zilla-Plugin-DualBuilders',
  'EXE_FILES' => [],
  'LICENSE' => 'perl',
  'NAME' => 'Dist::Zilla::Plugin::DualBuilders',
  'PREREQ_PM' => {
    'Dist::Zilla::Role::AfterBuild' => '3.101461',
    'Dist::Zilla::Role::InstallTool' => '3.101461',
    'Dist::Zilla::Role::PrereqSource' => '3.101461',
    'Moose' => '1.03',
    'Moose::Util::TypeConstraints' => '1.01'
  },
  'VERSION' => '1.001',
  'test' => {
    'TESTS' => 't/*.t'
  }
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.56) } ) {
  my $br = delete $WriteMakefileArgs{BUILD_REQUIRES};
  my $pp = $WriteMakefileArgs{PREREQ_PM};
  for my $mod ( keys %$br ) {
    if ( exists $pp->{$mod} ) {
      $pp->{$mod} = $br->{$mod} if $br->{$mod} > $pp->{$mod};
    }
    else {
      $pp->{$mod} = $br->{$mod};
    }
  }
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



